ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitter.scala

```diff
@@ -303,31 +303,41 @@ private[kvutils] class TransactionCommitter(
   }
 
   @nowarn("msg=deprecated")
-  private def setDedupEntry(
+  private[transaction] def setDedupEntry(
       commitContext: CommitContext,
       transactionEntry: DamlTransactionEntrySummary,
   )(implicit loggingContext: LoggingContext): Unit = {
     val (_, config) = getCurrentConfiguration(defaultConfig, commitContext)
-    val until = transactionEntry.submitterInfo.getDeduplicationPeriodCase match {
+    val commandDedupBuilder = DamlCommandDedupValue.newBuilder
+    transactionEntry.submitterInfo.getDeduplicationPeriodCase match {
       case DamlSubmitterInfo.DeduplicationPeriodCase.DEDUPLICATE_UNTIL =>
-        transactionEntry.submitterInfo.getDeduplicateUntil
-      case DamlSubmitterInfo.DeduplicationPeriodCase.DEDUPLICATION_DURATION |
-          DamlSubmitterInfo.DeduplicationPeriodCase.DEDUPLICATION_OFFSET |
-          DamlSubmitterInfo.DeduplicationPeriodCase.DEDUPLICATIONPERIOD_NOT_SET =>
-        Conversions.buildTimestamp(
-          transactionEntry.submissionTime
-            .add(config.maxDeduplicationTime)
-            .add(config.timeModel.minSkew)
+        commandDedupBuilder.setDeduplicatedUntil(transactionEntry.submitterInfo.getDeduplicateUntil)
+      case DamlSubmitterInfo.DeduplicationPeriodCase.DEDUPLICATION_DURATION =>
+        commandDedupBuilder.setDeduplicatedUntil(
+          Conversions.buildTimestamp(
+            transactionEntry.submissionTime
+              .add(
+                Conversions.parseDuration(transactionEntry.submitterInfo.getDeduplicationDuration)
+              )
+              .add(config.timeModel.minSkew)
```

Comment:
Why is the minimum skew used for these calculations? I would expect the maximum.

Reply:
Per design doc, I assume it;s the min skew we want to account for to be withing boundries